internal/mapping.go

```diff
@@ -40,8 +40,12 @@ func GetSpannerTable(conv *Conv, srcTable string) (string, error) {
 	if srcTable == "" {
 		return "", fmt.Errorf("bad parameter: table string is empty")
 	}
-	// Once computed, return cached result.
-	if sp, found := conv.ToSpanner[srcTable]; found {
+
+	tableId, err := GetTableIdFromName(conv, srcTable)
+	if err != nil {
+		return "", err
```

Comment:
We found the `srcTable` name before calling the `GetSpannerTable` to pass into it using `tableId` and now we using `srcTable` to reverse lookup the tableId.
Instead of doing that, we should pass in the `tableId` to the `GetSpannerTable` itself.

Since `tableId` : `tableName` mapping is always 1:1 and deterministic, you can also consider using something like a `BiMap`. This will simplify forward and reverse lookups significantly.

Reply:
Avoid reversed lookup by passing tableId as parameter. Please review it.